css/tokens/index.js

```diff
@@ -34,39 +28,127 @@ async function createTokens() {
+				const filePath = line.replace('@import', '').replaceAll(`'`, '').replace(';', '').trim();
+				arr.push(path.join(filePathStem, filePath));
+			}
+			return arr;
+		}, filePaths);
+		return lines;
+	} catch (err) {
+		throw err;
+	}
+}
+
+/**
+ *
+ * @param {import('./types').SassExportCollection} collection collection of grouped tokens to be sorted
+ * @returns {import('./types').SassExportCollection}
  */
-function collectTokenSources(rendered) {
-	const collection = rendered.stats.includedFiles
-		.map((/** @type {string} */ file) => {
-			return { name: getFinalFilePart(file), location: getAtlasFilePath(file) };
-		})
-		.filter((/** @type {{ name: string; location: string; }} */ x) => x.name !== 'index')
-		.reduce(
+function getSortedOrder(collection) {
+	return Object.fromEntries(Object.entries(collection).sort());
+}
+
+/**
+ *
+ * @param {string[]} paths token file paths
+ */
+// Print a warning if a token file lacks the required sass-export-section comments.
+function checkFileComments(paths) {
+	paths.forEach(path => {
+		try {
+			fs.readFile(path, 'utf8', function read(err, result) {
+				if (err) {
+					throw err;
+				}
+				if (!result.includes('@sass-export-section')) {
+					console.log(`Warning: ${path} is missing @sass-export-section annotations.`);
+				}
+			});
+		} catch (err) {
+			throw new Error(`Problem reading token files: ${err}`);
+		}
+	});
+}
+
+/**
+ *
+ * @param {import('./types').SassExportTokens} tokens raw token data by group
+ * @returns {import('./types').SassExportCollection}
+ */
+function collectTokens(tokens) {
+	/** @type {import('./types').SassExportCollection} */
+	const collection = {};
+	for (const [parent, tokenValues] of Object.entries(tokens)) {
+		//Currently using sass-export-section annotations in the token files for grouping.
+		//Tokens without annotations will be combined in the variables array.
+		if (parent === 'variables') continue;
+
+		const collectedValues = tokenValues.reduce(
 			(
-				/** @type {{ [x: string]: any; }} */ all,
-				/** @type {{ name: string | number; }} */ source
+				/** @type {import('./types').SassExportCollection } */ all,
+				/** @type {import('./types').SassExportTokenItem} */ current
 			) => {
-				all[source.name] = {
-					...source,
-					tokens: {}
+				const tokenName = current.name;
+
+				/** @type {string | import('./types').SassExportTokenNestedItem} */
+				const values = current.mapValue
+					? { ...getNestedTokens(current) }[tokenName]
+					: current.compiledValue;
+
+				all[parent] = {
+					...all[parent],
+					[tokenName]: values
 				};
 				return all;
 			},
 			{}
 		);
+		collection[parent] = {
+			name: parent,
+			location: `/css/src/tokens/${parent}.scss`,
+			tokens: collectedValues[parent]
+		};
+	}
 	return collection;
 }
 
+/**
+ *
+ * @param {import('./types').SassExportTokenItem} child
+ * @returns {{[key: string]: import('./types').SassExportTokenNestedItem}}
+ */
+function getNestedTokens(child) {
```

Comment:
Placing comment here only because I'm not sure where else to do so. Seeing some extra spaces in sass arrays. Is that us or sass-export?

![image](https://user-images.githubusercontent.com/30843002/128909309-6e8332bd-5a40-4c3d-9060-8ca09c6853bb.png)

Reply:
It's sass-export, add logic to remove extra spacing.